subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -156,67 +201,72 @@ do_firmware_transfer_reply_cb(const struct zoap_packet *response,
 	u8_t *payload;
 	struct zoap_packet *check_response = (struct zoap_packet *)response;
 	lwm2m_engine_set_data_cb_t callback;
-
-	SYS_LOG_DBG("TRANSFER REPLY");
+	u8_t resp_code;
+
+	/* Check response code from server. Expecting (2.05) */
+	resp_code = zoap_header_get_code(check_response);
+	if (resp_code != ZOAP_RESPONSE_CODE_CONTENT) {
+		SYS_LOG_ERR("Unexpected response from server: %d.%d",
+			    ZOAP_RESPONSE_CODE_CLASS(resp_code),
+			    ZOAP_RESPONSE_CODE_DETAIL(resp_code));
+		lwm2m_firmware_set_update_result(RESULT_CONNECTION_LOST);
+		return -ENOENT;
+	}
 
 	ret = zoap_update_from_block(check_response, &firmware_block_ctx);
 	if (ret < 0) {
 		SYS_LOG_ERR("Error from block update: %d", ret);
+		lwm2m_firmware_set_update_result(RESULT_INTEGRITY_FAILED);
 		return ret;
 	}
 
-	/* TODO: Process incoming data */
+	/* Reach last block if transfer_offset equals to 0 */
+	transfer_offset = zoap_next_block(check_response, &firmware_block_ctx);
```

Comment:
You are correct.
But we should remove the second one (the one inside the if (payload_len > 0))
Will update accordingly.